stub-runner/stub-runner/src/main/groovy/com/ofg/stub/server/AvailablePortScanner.groovy

```diff
@@ -7,38 +7,50 @@ import groovy.util.logging.Slf4j
 @Slf4j
 class AvailablePortScanner {
 
+    public static final int MAX_RETRY_COUNT = 1000
     private final int minPortNumber
     private final int maxPortNumber
-    private int portToScan
 
     AvailablePortScanner(int minPortNumber, int maxPortNumber) {
-        this.portToScan = minPortNumber
         this.minPortNumber = minPortNumber
         this.maxPortNumber = maxPortNumber
     }
 
-    int nextAvailablePort() {
-        ServerSocket socket = null
-        while (portToScan <= maxPortNumber) {
+    void tryToExecuteWithFreePort(Closure closure) {
```

Comment:
I love the idea behind `tryToExecuteWithFreePort`. However can it accept  `Closure<V>` and return `V`? This way it can return e.g.:

```java
    @Bean(destroyMethod = 'shutdownServer')
    HttpMockServer httpMockServer(AvailablePortScanner availablePortScanner) {
        return availablePortScanner.tryToExecuteWithFreePort { int availablePort ->
            HttpMockServer server = new HttpMockServer(availablePort)
            server.start()
            return server
        }
    }
```

As it is right now it must close over outer scope and rely on side effects, thus it's harder to test. But the idea itself is brilliant.